app/models/color_scheme.rb

```diff
@@ -343,8 +345,9 @@ def self.hex_cache
   after_save_commit :publish_discourse_stylesheet, unless: :skip_publish
   after_save_commit :dump_caches
   after_destroy :dump_caches
+  after_destroy :destroy_remote_original
   belongs_to :theme
-  belongs_to :base_scheme, class_name: "ColorScheme"
+  belongs_to :base_scheme, -> { unscope(where: :remote_copy) }, class_name: "ColorScheme"
```

Comment:
I think it'll be helpful to introduce a `scope :base_schemes, -> { unscope(where: :remote_copy) }` so that we can use the scope here and also in the `destroy_remote_original` callback.